src/session.cpp

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2021 YADRO
+
+#include <dbus.hpp>
+#include <phosphor-logging/elog-errors.hpp>
+#include <phosphor-logging/log.hpp>
+#include <session.hpp>
+
+#include <iostream>
+
+namespace obmc
+{
+namespace session
+{
+
+using namespace phosphor::logging;
+
+void SessionItem::setSessionMetadata(std::string username,
+                                     std::string remoteIPAddr)
+{
+    this->adjustSessionOwner(username);
+    if (remoteIPAddr.empty())
+    {
+        throw InvalidArgument();
+    }
+    this->remoteIPAddr(remoteIPAddr);
+}
+
+void SessionItem::adjustSessionOwner(const std::string& userName)
+{
+    using DBusGetObjectOut = std::map<std::string, std::vector<std::string>>;
+
+    constexpr const std::array userObjectIfaces = {
+        "xyz.openbmc_project.User.Attributes"};
+    auto userObjectPath = "/xyz/openbmc_project/user/" + userName;
+
+    DBusGetObjectOut getUserObject;
+    auto callMethod =
+        bus.new_method_call("xyz.openbmc_project.ObjectMapper",
+                            "/xyz/openbmc_project/object_mapper",
+                            "xyz.openbmc_project.ObjectMapper", "GetObject");
```

Comment:
All of these constants are defined in dbus.hpp.
Don't you think that this method is also worth putting there?

Reply:
Yes, I missed this. Sure, all this stuff should be constants from dbus.hpp.

Reply:
Fixed